lib/internal/http2/compat.js

```diff
@@ -668,22 +666,18 @@ class Http2ServerResponse extends Stream {
     if (chunk !== null && chunk !== undefined)
       this.write(chunk, encoding);
 
-    const isFinished = this.finished;
-    state.headRequest = stream.headRequest;
     state.ending = true;
 
     if (typeof cb === 'function') {
-      if (isFinished)
+      // TODO(ronag): ERR_STREAM_ALREADY_FINISHED
+      if (stream.writableFinished)
         this.once('finish', cb);
       else
         stream.once('finish', cb);
     }
 
     if (!stream.headersSent)
       this.writeHead(this[kState].statusCode);
```

Comment:
NOTE: `writeHead` will end the stream so we don't need to call `end()` explicitly.